extensions/amp-script/amp-script.md

```diff
@@ -138,6 +138,51 @@ Will be reflected on the page as a new child of the `amp-script` element:
 
 Under the hood, `amp-script` uses [@ampproject/worker-dom](https://github.com/ampproject/worker-dom/). For design details, see the ["Intent to Implement" issue](https://github.com/ampproject/amphtml/issues/13471).
 
+### Advanced usages
+
+#### Passing data between amp-script and the page
+
+`amp-script` supports getting and setting [`amp-state`](https://amp.dev/documentation/components/amp-bind/#initializing-state-with-amp-state) JSON via JavaScript.
+
+This enables advanced interactions between `amp-script` and other AMP elements on the page via `amp-bind` [bindings](https://amp.dev/documentation/components/amp-bind/#bindings).
+
+[tip type="default"]
+`AMP.setState()` requires the [`amp-bind`](https://amp.dev/documentation/components/amp-bind) extension to be installed.
+[/tip]
+
+```js
+/**
+ * Deep-merges `json` into the current amp-state.
+ * @param {!Object} json A JSON object e.g. must not contain circular references.
+ */
+AMP.setState(json) {}
+
+/**
+ * Asynchronously returns amp-state.
+ * @param {string=} expr An optional JSON expression string e.g. "foo.bar".
+ * @return {!Promise<!Object>}
+ */
+AMP.getState(expr) {}
+```
+
+##### Example with WebSocket and AMP.setState()
```

Comment:
Does this also work with an JavaScript file imported through a URL?

Reply:
Yes, it's just easier to show examples with inline scripts.

Reply:
Totally understandable. I think this is fine. :)